File: Question 0070 - Climbing Stairs/src/Solution.java
public class Solution
{
    private int solve(int n)
    {
        int prev2 = 1;
        int prev1 = 1;

        int ans = 0;
        for (int i = 2; i <= n; i++) {
            ans = prev1 + prev2;
            prev2 = prev1;
            prev1 = ans;
        }

        return ans;
    }

    public int climbStairs(int n)
    {
        if (n == 0 || n == 1) return 1;
        return solve(n);
    }
}

File: Question 0070 - Climbing Stairs/src/Solution.cpp
#include<bits/stdc++.h>
using namespace std;

int solve(int n, vector<int>& dp)
{
    //base case
    if(n < 0) return 0;
    if(n == 0 || n == 1) return 1;
    if(dp[n] != -1) return dp[n];

    return dp[n] = solve(n-1, dp) + solve(n-2, dp);
}

class Solution 
{
public:
    int climbStairs(int n) 
    {
        vector<int> dp(n+1, -1);
        return solve(n, dp);
    }
};